bazel/kotlin_test.bzl

```diff
@@ -18,11 +18,17 @@ load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_test")
 # )
 #
 def envoy_mobile_kt_test(name, srcs, deps = []):
+    # This is to work around the issue where we have specific implementation functionality which
+    # we want to avoid comsumers to use but we want to unit test
+    dep_srcs = []
+    for dep in deps:
+        dep_srcs.append(dep + "_srcs")
```

Comment:
Why is this necessary if the file in question is being included in the dependency?

Reply:
In Kotlin, there's a concept of `internal` which allows for module level visibility.

The problem here is that in Bazel, we're using the library target as a dependency (so the test itself doesn't get visibility for `internal` entities). The reason this works is because we're basically leveraging the workaround in a [previous pr](https://github.com/lyft/envoy-mobile/pull/264) and use those defined `filegroup`s as source files in the test. Since it's now sources within the test target, we are able to access `internal` entities.

The other concept that I would like to call out is that we want to keep this entity (the mapper) an `internal` visible entity because we don't necessarily want consumers to use it. We also don't want to make it strictly private since that would make testing this behavior more complex (we'll have to do some integration black box testing or something along those lines).